language_generators/scala-defref-extractor/src/main/scala/io/bazeltools/buildgen/scaladefref/ScalaSourceEntityExtractor.scala

```diff
@@ -482,17 +545,61 @@ object ScalaSourceEntityExtractor {
     processScopeTree(ss.tree)
   }
 
+  def log(s: String): Unit = {
+    require(s ne null)
+    // println(s)
+  }
+
+  def treeToNames(tpe: scala.meta.Tree): List[Name] = {
+    import scala.meta._
+    tpe match {
+      case Defn.Def(mods, name, tparams, paramss, optType, body) =>
+        log(
+          s"treeToNames:  Defn.Def(mods:$mods, name: $name, tparams: $tparams, paramss: $paramss, optType: $optType, body: $body)"
+        )
+        optType.toList.map(typeToNames).flatten ++
+          paramss.flatMap(_.map(treeToNames)).flatten
+      case Term.Param(mods, name, optType, optTerm) =>
+        log(s"treeToNames: Term.Param($mods, $name, $optType, $optTerm)")
+        optType.toList.map(typeToNames).flatten
+      case Init(tpe, name, terms) =>
+        log(s"Init($tpe, $name, $terms)")
+        typeToNames(tpe)
+
+      case Defn.Val(mods, pats, optType, term) =>
+        log(
+          s"treeToNames: Val(mods: $mods, pats: $pats, optType: $optType, term: $term)"
+        )
+        optType.toList.flatMap(typeToNames)
+      case _ => Nil
+
+      // case t =>
+      // sys.error(s"unknown: ${t.getClass} $t")
+    }
+  }
+  def typeToNames(tpe: scala.meta.Type): List[Name] = {
+    import scala.meta._
+    tpe match {
+      case tn @ Type.Name(_) =>
```

Comment:
nit: `tn :: Nil` is possibly faster since it is fewer allocations. `List(tn)` does `List.apply(Seq(tn))` basically.